src/YesSql.Core/Services/TypeService.cs

```diff
@@ -8,6 +10,15 @@ public class TypeService : ITypeService
     {
         private readonly ConcurrentDictionary<Type, string> typeNames = new ConcurrentDictionary<Type, string>();
 
+        public IEnumerable<Type> Keys { get { return typeNames.Keys; } }
+
+        public IEnumerable<string> Values { get { return typeNames.Values; } }
+
+        public Type ReverseLookup(string value)
+        {
+            return typeNames.FirstOrDefault(t => t.Value == value).Key;
```

Comment:
You actually need two `ConcurrentDictionary`. And when a vale is set on the indexers, just set the two dictionaries.

Reply:
Two dictionaries makes sense.  That would guarantee uniqueness in both directions.